kubespawner/spawner.py

```diff
@@ -771,6 +795,24 @@ def get_pod_manifest(self):
             extra_containers=self.singleuser_extra_containers
         )
 
+    def get_cpu_limit(self):
```

Comment:
Why all these getter methods? One of the nice things about Python is that you don't need getter methods to get all of the benefits of getters. Subclasses can define properties or extend the traitlets here without needing to add any methods on the base class.

Reply:
I was under the impression that we need these to be present to facilitate overriding as in https://gist.github.com/foxish/3407150729b106a05811d672d905c27b. If we can performing overriding of that kind without really adding those getters, that would be ideal. Can you elaborate a bit more on the mechanism by which one can do that?

Reply:
@foxish you can do so by overriding start like https://github.com/berkeley-dsep-infra/datahub/blob/staging/datahub/config.yaml#L12